src/dirnode.cpp

```diff
@@ -920,18 +920,36 @@ void RDirNode::logic(float dt) {
     since_last_node_change += dt;
 }
 
-void RDirNode::drawDirName(FXFont& dirfont) const{
+void RDirNode::drawDirName(FXFont& dirfont) {
     if(parent==0) return;
     if(gGourceSettings.hide_dirnames) return;
     if(gGourceSettings.dir_name_depth > 0 && gGourceSettings.dir_name_depth < (depth-1)) return;
 
     if(!gGourceSettings.highlight_dirs && since_last_node_change > 5.0) return;
 
     float alpha = gGourceSettings.highlight_dirs ? 1.0 : std::max(0.0f, 5.0f - since_last_node_change) / 5.0f;
+    dirfont.setAlpha(alpha);
 
-    vec2 label_pos = spline.getLabelPos();
+    vec2 new_label_delta(0.0f, 0.0f);
+    if (parent->getProjectedPos().y > projected_pos.y) {
+        // alignBottom
+        new_label_delta.y = dirfont.getHeight();
+    }
+    if (parent->getProjectedPos().x < projected_pos.x) {
+        // alignRight
+        new_label_delta.x = dirfont.getWidth(path_token);
+    }
 
-    dirfont.setAlpha(alpha);
+    if (label_delta != new_label_delta) {
+        vec2 label_diff = new_label_delta - label_delta;
+        if (glm::length(label_diff) < 1) {
+            label_delta = new_label_delta;
+        } else {
+            label_delta += label_diff / 10.0f;
```

Comment:
According to https://github.com/acaudwell/Gource/pull/174#discussion_r220402582
1. The "min" is done by line 945
2. I think we do not need dt here. Only the label_delta is animated. label_pos is still the old one resulting from a calculation where dt is considered. This is a sub-animation to adjust the label position where we can ignore dt (?).
3. / 10.0f is an easy function to slow down the animation when reaching the endpoint. As bigger the distance to go as faster is the animation at the beginning. Maybe we need to talk about a "Easing.type" like definied in http://doc.qt.io/qt-5/qml-qtquick-animator.html#easing.type-prop ?